### PR TITLE
New docker overhaul using Docker Compose

### PR DESCRIPTION
**Pull request by heliocastro**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (Copyright BMW CarIT GmbH 2021)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

* Using multi stage build with buildkit
* Download dependencies externally to reduce docker rebuild
* Build thrift and clucene as a separate entityies
*  Postgres and Couchdb now are containers using docker-compose
* Base image is not the final image, only with base Liferay and installs
* SW360 is built as last stage, allowing to easily change the code and
rebuild containers for test purposes

Must:
- [ ] SW360 must start as expected in pre-configuration stage

### CODE REVIEW

**Comment by mcjaeger** [comment]
looks like someone has no valid ssl certificates (search.maven...), it brings:
```
cara:sw360 sam$  ./docker_build.sh -f
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: certificate has expired
```
adding `-k`to the curl command which downloads the dependencies solves it.

**Comment by heliocastro** [comment]
Fixing. Thanks

**Comment by heliocastro** [comment]
> looks like someone has no valid ssl certificates (search.maven...), it brings:
> 
> ```
> cara:sw360 sam$  ./docker_build.sh -f
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> curl: (60) SSL certificate problem: certificate has expired
> ```
> 
> adding `-k`to the curl command which downloads the dependencies solves it.

I will fix, but usually means that ca-certificates are missing. Very strange

File: README_DOCKER.md [hunk 1]
```diff
@@ -1,24 +1,136 @@
...
+    PROXY_HTTPS_HOST=<your_https_proxy_ip>
+    PROXY_PORT=<your_port>
+    ```
+
+## Running the image
+
+* Run the resulting image:
+
+    ```sh
+    docker-compose -env-file scripts/docker-config/default.docker.env up
```
Comment (mcjaeger) [inline README_DOCKER.md:58]: small typo, should be `--env-file`

**Review by mcjaeger — APPROVED** [review]
some comments were outdated, but github was not able to find them (therefore, re-request review)

**Pull request merged**